syntax = "proto3";
package esw;

option java_package = "cz.esw.serialization.proto";
option java_multiple_files = true;

message PBInfo {
    int64 id = 1;
    string measurerName = 2;
    int64 timestamp = 3;
}

//cpp sending java recieveing part

message PBAvgData {
    PBInfo info = 1;
    double download = 2;
    double upload = 3;
    double ping = 4;
}

message PBAvgDataArray {
    repeated PBAvgData dataArray = 1;
}

//java sending cpp recieveing part

message PBMeasurement {
    PBInfo info = 1;
    repeated double download = 2;
    repeated double upload = 3;
    repeated double ping = 4;
}

message PBMeasurementArray {
    repeated PBMeasurement datasets = 1;
}